### PR TITLE
Adds test collections filtering

### PR DESCRIPTION
**Pull request by Mpdreamz**
> build integrate 5.0.0 readonly getindexapi

Also fixes SkipMethod being too greedy ReadOnly cluster was only
yielding 14 tests.

### CODE REVIEW

File: build.bat [hunk 1]
```diff
@@ -48,6 +49,7 @@ IF /I "%1"=="release" (
 IF /I "%1%"=="integrate" (
 	IF NOT [%2]==[] (set ESVERSIONS="%2")
 	IF NOT [%3]==[] (set NEST_INTEGRATION_CLUSTER="%3")
+	IF NOT [%4]==[] (set NEST_TEST_FILTER="%4")
```
Comment (russcam) [inline build.bat]: AW YEAH 👍

File: src/Tests/Framework/XUnitPlumbing/IntegrationTestDiscoverer.cs [hunk 2]
```diff
@@ -15,8 +16,11 @@ public IntegrationTestDiscoverer(IMessageSink diagnosticMessageSink)
 		protected override bool SkipMethod(ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo factAttribute)
 		{
 			var classOfMethod = Type.GetType(testMethod.TestClass.Class.Name, true, true);
-			var method = classOfMethod.GetMethod(testMethod.Method.Name);
-			var collectionType = testMethod.TestClass?.TestCollection?.CollectionDefinition?.Name;
+			var method = classOfMethod.GetMethod(testMethod.Method.Name, BindingFlags.FlattenHierarchy | BindingFlags.NonPublic | BindingFlags.Public)
```
Comment (russcam) [inline src/Tests/Framework/XUnitPlumbing/IntegrationTestDiscoverer.cs]: my bad, good catch 😲

**Comment by Mpdreamz** [comment]
I'm assuming by your comments this PR is ok @russcam, pulling it in because I need it on another branch :smile:

**Comment by russcam** [comment]
++ I needed it on two branches today :smile:

**Comment by Mpdreamz** [comment]
ported to `2.x` and `5.x`

**Pull request merged**